pybwa/libbwamem.pyi

```diff
@@ -19,39 +19,39 @@ class BwaMemMode(enum.Enum):
-        match_score: int | None = None,
-        mismatch_penalty: int | None = None,
-        minimum_score: int | None = None,
-        unpaired_penalty: int | None = None,
+        band_width: int | None = 100,
+        match_score: int | None = 1,
+        mismatch_penalty: int | None = 4,
+        minimum_score: int | None = 30,
+        unpaired_penalty: int | None = 17,
         n_threads: int | None = None,
-        skip_pairing: bool | None = None,
-        output_all_for_fragments: bool | None = None,
-        interleaved_paired_end: bool | None = None,
-        short_split_as_secondary: bool | None = None,
-        skip_mate_rescue: bool | None = None,
-        soft_clip_supplementary: bool | None = None,
-        with_xr_tag: bool | None = None,
-        query_coord_as_primary: bool | None = None,
-        keep_mapq_for_supplementary: bool | None = None,
-        with_xb_tag: bool | None = None,
-        max_occurrences: int | None = None,
-        off_diagonal_x_dropoff: int | None = None,
-        ignore_alternate_contigs: bool | None = None,
-        internal_seed_split_factor: float | None = None,
-        drop_chain_fraction: float | None = None,
-        max_mate_rescue_rounds: int | None = None,
-        min_seeded_bases_in_chain: int | None = None,
-        seed_occurrence_in_3rd_round: int | None = None,
-        xa_max_hits: int | tuple[int, int] | None = None,
-        xa_drop_ratio: float | None = None,
-        gap_open_penalty: int | tuple[int, int] | None = None,
-        gap_extension_penalty: int | tuple[int, int] | None = None,
-        clipping_penalty: int | tuple[int, int] | None = None,
+        skip_pairing: bool | None = False,
+        output_all_for_fragments: bool | None = False,
+        interleaved_paired_end: bool | None = False,
+        short_split_as_secondary: bool | None = False,
+        skip_mate_rescue: bool | None = False,
+        soft_clip_supplementary: bool | None = False,
+        with_xr_tag: bool | None = False,
+        query_coord_as_primary: bool | None = False,
+        keep_mapq_for_supplementary: bool | None = False,
+        with_xb_tag: bool | None = False,
+        max_occurrences: int | None = 500,
+        off_diagonal_x_dropoff: int | None = 100,
+        ignore_alternate_contigs: bool | None = False,
+        internal_seed_split_factor: float | None = 1.5,
+        drop_chain_fraction: float | None = 0.50,
+        max_mate_rescue_rounds: int | None = 50,
+        min_seeded_bases_in_chain: int | None = 0,
+        seed_occurrence_in_3rd_round: int | None = 20,
+        xa_max_hits: int | tuple[int, int] | None = (5, 200),
+        xa_drop_ratio: float | None = 0.80,
+        gap_open_penalty: int | tuple[int, int] | None = 6,
+        gap_extension_penalty: int | tuple[int, int] | None = 1,
+        clipping_penalty: int | tuple[int, int] | None = 5,
         threads: int | None = None,
-        chunk_size: int | None = None,
+        chunk_size: int | None = 10_000_000,
```

Comment:
Same comments as for `BwaAlnOptions` 🙂 

1. Can these now all be typed as `T` instead of `T | None`?
2. Can `mode` and `threads` be given defaults?

Also, what is the difference between `threads` and `n_threads`?

Reply:
`mode` is special, in that it is `None` (or null pointer in bwa) by default.

`threads` was added in https://github.com/fulcrumgenomics/pybwa/pull/28 to have the same name as `bwaaln`.  I kept `n_threads` as not to break the API.  I am now tracking this at #94.